compiler/resolution/lowerIterators.cpp

```diff
@@ -1290,8 +1302,16 @@ expandIteratorInline(ForLoop* forLoop) {
     // the yielded index for the iterator formal.
     expandBodyForIteratorInline(forLoop, ibody, index);
 
-    collect_asts(ibody, asts);
 
+    if (iterator->throwsError()) {
+      // For an inlined iterator, we don't want to track the error
+      // argument through the iterator class. Instead of setting
+      // a formal error argument, set an enclosing error variable.
+      collect_asts(ibody, asts);
+      replaceErrorArgumentWithEnclosingError(iterator, asts);
+    }
+
+    collect_asts(ibody, asts);
     replaceIteratorFormalsWithIteratorFields(iterator, ic, asts);
```

Comment:
replaceErrorArgumentWithEnclosingError does change the set of SymExprs, since it creates a CallExpr and adds dummy error arguments. But perhaps it only adds SymExprs that replaceIteratorFormalsWithIteratorFields does not need to concern itself with. But in that event I think it's better to err on clarity/less potential bugs - in case that changes - than to optimize it.

Anyway I'll change them to use a vector of SymExprs. This change involves changes to other functions here, but it seems like an improvement... It'll be in a separate commit though so you can have a look.